react/features/prejoin/components/dialogs/JoinByPhoneDialog.js

```diff
@@ -0,0 +1,245 @@
+// @flow
+
+import React, { Component } from 'react';
+import {
+    dialOut as dialOutAction,
+    joinConferenceWithoutAudio as joinConferenceWithoutAudioAction,
+    openDialInPage as openDialInPageAction
+} from '../../actions';
+import { connect } from '../../../base/redux';
+import CallingDialog from './CallingDialog';
+import DialOutDialog from './DialOutDialog';
+import DialInDialog from './DialInDialog';
+import { getDialOutStatus, getFullDialOutNumber } from '../../functions';
+import {
+    getConferenceId,
+    getDefaultDialInNumber,
+    updateDialInNumbers
+} from '../../../invite';
+
+type Props = {
+
+    /**
+     * The number to call in order to join the conference.
+     */
+    dialInNumber: string,
+
+    /**
+     * The status of the call when the meeting calss the user.
+     */
+    dialOutStatus: string,
+
+    /**
+     * The action by which the meeting calls the user.
+     */
+    dialOut: Function,
+
+    /**
+     * The number the conference should call.
+     */
+    dialOutNumber: string,
+
+    /**
+     * Fetches conference dial in numbers & conference id
+     */
+    fetchConferenceDetails: Function,
+
+    /**
+     * Joins the conference without audio.
+     */
+    joinConferenceWithoutAudio: Function,
+
+    /**
+     * Closes the dialog.
+     */
+    onClose: Function,
+
+    /**
+     * Opens a web page with all the dial in numbers.
+     */
+    openDialInPage: Function,
+
+    /**
+     * The pin of the conference used when joining a meeting by phone.
+     */
+    pin: string,
+};
+
+type State = {
+
+    /**
+     * The dialout call is ongoing, 'CallingDialog' is shown;
+     */
+    isCalling: boolean,
+
+    /**
+     * If should show 'DialOutDialog'.
+     */
+    showDialOut: boolean,
+
+    /**
+     * If should show 'DialInDialog'.
+     */
+    showDialIn: boolean,
+}
+
+/**
+ * This is the dialog shown when a user wants to join with phone audio.
+ */
+class JoinByPhoneDialog extends Component<Props, State> {
+    /**
+     * Initializes a new {@code JoinByPhoneDialog} instance.
+     *
+     * @param {Props} props - The props of the component.
+     * @inheritdoc
+     */
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isCalling: false,
+            showDialOut: true,
+            showDialIn: false
+        };
+        this._dialOut = this._dialOut.bind(this);
```

Comment:
It's not a must have, but it's nice to leave an empty line between these blocks (e.g. state setting and this binding)

Reply:
Done.